modules/matchlist/matchlist.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package matchlist
+
+import (
+	"github.com/gobwas/glob"
+)
+
+// Matchlist represents a black or white list
+type Matchlist struct {
+	rules     []string
+	ruleGlobs []glob.Glob
+}
+
+// NewMatchlist creates a new black or white list
+func NewMatchlist(rules ...string) (*Matchlist, error) {
+	list := Matchlist{
+		rules:     rules,
+		ruleGlobs: make([]glob.Glob, 0, len(rules)),
+	}
+
+	for _, rule := range list.rules {
+		rg, err := glob.Compile(rule)
```

Comment:
```suggestion
		rg, err := glob.Compile(rule,[]string{".","/"})
```
I think that separators should work better if specified, so the tokenization is aware of the different parts of the glob.